blob/src/main/java/io/crate/blob/v2/BlobIndex.java

```diff
@@ -65,24 +71,71 @@ public static String stripPrefix(String indexName) {
 
     private final Map<Integer, BlobShard> shards = new ConcurrentHashMap<>();
     private final Path globalBlobPath;
+    private final ESLogger logger;
 
-
-    BlobIndex(@Nullable Path globalBlobPath) {
+    BlobIndex(ESLogger logger, @Nullable Path globalBlobPath) {
         this.globalBlobPath = globalBlobPath;
+        this.logger = logger;
     }
 
     void createShard(IndexShard indexShard) {
         shards.put(indexShard.shardId().id(), new BlobShard(indexShard, globalBlobPath));
     }
 
     BlobShard removeShard(ShardId shardId) {
+        Path blobRoot = null;
         BlobShard shard = shards.remove(shardId.id());
         if (shard != null) {
+            if (shards.size() == 0) {
+                blobRoot = this.retrieveBlobRootDir(shard);
+            }
             shard.deleteShard();
+            if (blobRoot != null) {
```

Comment:
is this split into two conditions necessary? I think retrieveBlobRootDir can still be called after `shard.deleteShard()`